### PR TITLE
[Docs] Add maintenance plan

### PR DESCRIPTION
**Pull request by zhouzaida**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add maintenance plan and print warning info when importing mmcv.

## Modification

Update the following files:

- README.md
- README_zh-CN.md
- docs/en/compatibility.md
- docs/zh_cn/compatibility.md
- mmcv/__init__.py

## BC-breaking (Optional)

No

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,16 @@
 
 English | [简体中文](README_zh-CN.md)
 
+## Highlights
+
+The OpenMMLab team released a new generation of training engine [MMEngine](https://github.com/open-mmlab/mmengine) at the World Artificial Intelligence Conference on September 1, 2022. It is a foundational library for training deep learning models. Compared with MMCV, it provides a universal and powerful runner, an open architecture with a more unified interfaces, and a more customizable training process.
+
+At the same time, MMCV released [2.x](https://github.com/open-mmlab/mmcv/tree/2.x) release candidate version and will release 2.x officially version on January 1, 2023.
+
+In version 2.x, it removed components related to the training process and added a data transformation module. Also, starting from 2.x, rename the package names **mmcv** to **mmcv-lite** and **mmcv-full** to **mmcv**. For details, see [Compatibility Documentation](docs/en/compatibility.md).
```
Comment (HAOCHENYE) [inline README.md:40]:
```suggestion
In version 2.x, it removed components related to the training process and added a data transformation module. Also, starting from 2.x, MMCV renamed the package names **mmcv** to **mmcv-lite**, and **mmcv-full** to **mmcv**. For details, see [Compatibility Documentation](docs/en/compatibility.md).
```

File: README_zh-CN.md [hunk 2]
```diff
@@ -31,6 +31,14 @@
 
 [English](README.md) | 简体中文
 
+## Highlights
+
+OpenMMLab 团队于 2022 年 9 月 1 日在世界人工智能大会发布了新一代训练引擎 [MMEngine](https://github.com/open-mmlab/mmengine)，它是一个用于训练深度学习模型的基础库。相比于 MMCV，它提供了更高级且通用的训练器、接口更加统一的开放架构以及可定制化程度更高的训练流程。
```
Comment (HAOCHENYE) [inline README_zh-CN.md:36]:
```suggestion
OpenMMLab 团队于 2022 年 9 月 1 日在世界人工智能大会发布了新一代训练引擎 [MMEngine](https://github.com/open-mmlab/mmengine)，它是一个用于训练深度学习模型的基础库。相比于 MMCV，它提供了更高级且通用的训练器、更加统一的接口、更加的开放架构，以及定制化程度更高的训练流程。
```

File: README_zh-CN.md [hunk 3]
```diff
@@ -31,6 +31,14 @@
...
+
+与此同时，MMCV 发布了 [2.x](https://github.com/open-mmlab/mmcv/tree/2.x) 预发布版本并将于 2023 年 1 月 1 日发布 2.x 正式版本。在 2.x 版本中，它删除了和训练流程相关的组件，并新增了数据变换模块。另外，从 2.x 版本开始，重命名包名 **mmcv** 为 **mmcv-lite** 以及 **mmcv-full** 为 **mmcv**。详情见[兼容性文档](docs/zh_cn/compatibility.md)。
```
Comment (HAOCHENYE) [inline README_zh-CN.md:38]:
```suggestion
与此同时，MMCV 发布了 [2.x](https://github.com/open-mmlab/mmcv/tree/2.x) 预发布版本，并将于 2023 年 1 月 1 日发布 2.x 正式版本。在 2.x 版本中，它删除了和训练流程相关的组件，并新增了数据变换模块。另外，从 2.x 版本开始，重命名包名 **mmcv** 为 **mmcv-lite** 以及 **mmcv-full** 为 **mmcv**。详情见[兼容性文档](docs/zh_cn/compatibility.md)。
```

File: docs/en/compatibility.md [hunk 4]
```diff
@@ -1,3 +1,53 @@
+### v2.0.0rc1
+
+The OpenMMLab team released a new generation of training engine [MMEngine](https://github.com/open-mmlab/mmengine) at the World Artificial Intelligence Conference on September 1, 2022. It is a foundational library for training deep learning models. Compared with MMCV, it provides a universal and powerful runner, an open architecture with a more unified interfaces, and a more customizable training process.
+
+At the same time, MMCV released [2.x](https://github.com/open-mmlab/mmcv/tree/2.x) release candidate version and will release 2.x officially version on January 1, 2023. In version 2.x, it has the following changes:
+
+(1) It removes the following components:
+
+- Removed `mmcv.fileio` module in PR [#2179](https://github.com/open-mmlab/mmcv/pull/2179) and used fileio module from mmengine wherever fileio is required.
```
Comment (HAOCHENYE) [inline docs/en/compatibility.md:9]:
```suggestion
- Removed `mmcv.fileio` module in PR [#2179](https://github.com/open-mmlab/mmcv/pull/2179) and using fileio module from mmengine wherever fileio is required.
```

File: docs/en/compatibility.md [hunk 5]
```diff
@@ -1,3 +1,53 @@
...
+- Removed four modules in PR [#2216](https://github.com/open-mmlab/mmcv/pull/2216), `mmcv.runner`, `mmcv.parallel`, `mmcv. engine` and `mmcv.device`
+- Removed all classes in `mmcv.utils` (eg `Config` and `Registry`) and large Some functions in PR [#2217](https://github.com/open-mmlab/mmcv/pull/2217) , only a few functions related to mmcv are reserved.
```
Comment (HAOCHENYE) [inline docs/en/compatibility.md:11]:
```suggestion
- Removed all classes in `mmcv.utils` (eg `Config` and `Registry`) and some large functions in PR [#2217](https://github.com/open-mmlab/mmcv/pull/2217) , only a few functions related to mmcv are reserved.
```

File: docs/en/compatibility.md [hunk 6]
```diff
@@ -1,3 +1,53 @@
...
+- Removed `mmcv.onnex`, `mmcv.tensorrt` modules and related functions in PR [#2225](https://github.com/open-mmlab/mmcv/pull/2225).
+
+(2) It adds the [`mmcv.transforms`](https://github.com/open-mmlab/mmcv/tree/2.x/mmcv/transforms) data transformation module.
+
+(3) It renamed the package name **mmcv** to **mmcv-lite**, **mmcv-full** renamed to **mmcv** in PR [#2235](https://github.com/open-mmlab/mmcv/pull/2235). Also, change the default value of the environment variable `MMCV_WITH_OPS` from 0 to 1.
```
Comment (HAOCHENYE) [inline docs/en/compatibility.md:16]: `renamed`, `adds`, `removes`. Should theres tenses should be unified?

File: docs/zh_cn/compatibility.md [hunk 7]
```diff
@@ -1,3 +1,53 @@
...
+（1）它删除了以下组件：
+
+- 在 PR [#2179](https://github.com/open-mmlab/mmcv/pull/2179) 中删除了 `mmcv.fileio` 模块并在需要使用 fileio 的地方使用 mmengine 中的 fileio 模块
+- 在 PR [#2216](https://github.com/open-mmlab/mmcv/pull/2216) 中删除了四个模块，分别是 `mmcv.runner`、`mmcv.parallel`、`mmcv.engine` 和 `mmcv.device`
+- 在 PR [#2217](https://github.com/open-mmlab/mmcv/pull/2217) 中删除了 `mmcv.utils` 中的所有类（例如 `Config` 和 `Registry`）和大部分函数，只保留少数和 mmcv 相关的函数
+- 在 PR [#2225](https://github.com/open-mmlab/mmcv/pull/2225) 中删除了 `mmcv.onnex`、`mmcv.tensorrt` 模块以及相关的函数
+
+（2）它新增了 [`mmcv.transforms`](https://github.com/open-mmlab/mmcv/tree/2.x/mmcv/transforms) 数据变换模块
+
+（3）它在 PR [#2235](https://github.com/open-mmlab/mmcv/pull/2235) 中将包名 **mmcv** 重命名为 **mmcv-lite**、 **mmcv-full** 重命名为 **mmcv**。此外，将环境变量 `MMCV_WITH_OPS` 的默认值从 0 改为 1
```
Comment (HAOCHENYE) [inline docs/zh_cn/compatibility.md:16]:
```suggestion
（1）删除了以下组件：

- 在 PR [#2179](https://github.com/open-mmlab/mmcv/pull/2179) 中删除了 `mmcv.fileio` 模块并在需要使用 fileio 的地方使用 mmengine 中的 fileio 模块
- 在 PR [#2216](https://github.com/open-mmlab/mmcv/pull/2216) 中删除了四个模块，分别是 `mmcv.runner`、`mmcv.parallel`、`mmcv.engine` 和 `mmcv.device`
- 在 PR [#2217](https://github.com/open-mmlab/mmcv/pull/2217) 中删除了 `mmcv.utils` 中的所有类（例如 `Config` 和 `Registry`）和大部分函数，只保留少数和 mmcv 相关的函数
- 在 PR [#2225](https://github.com/open-mmlab/mmcv/pull/2225) 中删除了 `mmcv.onnex`、`mmcv.tensorrt` 模块以及相关的函数

（2）新增了 [`mmcv.transforms`](https://github.com/open-mmlab/mmcv/tree/2.x/mmcv/transforms) 数据变换模块

（3）在 PR [#2235](https://github.com/open-mmlab/mmcv/pull/2235) 中将包名 **mmcv** 重命名为 **mmcv-lite**、 **mmcv-full** 重命名为 **mmcv**。此外，将环境变量 `MMCV_WITH_OPS` 的默认值从 0 改为 1
```

File: README.md [hunk 8]
```diff
@@ -31,6 +31,16 @@
 
 English | [简体中文](README_zh-CN.md)
 
+## Highlights
+
+The OpenMMLab team released a new generation of training engine [MMEngine](https://github.com/open-mmlab/mmengine) at the World Artificial Intelligence Conference on September 1, 2022. It is a foundational library for training deep learning models. Compared with MMCV, it provides a universal and powerful runner, an open architecture with a more unified interfaces, and a more customizable training process.
```
Comment (C1rN09) [inline README.md:36]:
```suggestion
The OpenMMLab team released a new generation of training engine [MMEngine](https://github.com/open-mmlab/mmengine) at the World Artificial Intelligence Conference on September 1, 2022. It is a foundational library for training deep learning models. Compared with MMCV, it provides a universal and powerful runner, an open architecture with a more unified interface, and a more customizable training process.
```

File: docs/en/compatibility.md [hunk 9]
```diff
@@ -1,3 +1,53 @@
+### v2.0.0rc1
+
+The OpenMMLab team released a new generation of training engine [MMEngine](https://github.com/open-mmlab/mmengine) at the World Artificial Intelligence Conference on September 1, 2022. It is a foundational library for training deep learning models. Compared with MMCV, it provides a universal and powerful runner, an open architecture with a more unified interfaces, and a more customizable training process.
```
Comment (C1rN09) [inline docs/en/compatibility.md:3]:
```suggestion
The OpenMMLab team released a new generation of training engine [MMEngine](https://github.com/open-mmlab/mmengine) at the World Artificial Intelligence Conference on September 1, 2022. It is a foundational library for training deep learning models. Compared with MMCV, it provides a universal and powerful runner, an open architecture with a more unified interface, and a more customizable training process.
```

File: docs/en/compatibility.md (hunk 4 above)
Comment (C1rN09) [inline docs/en/compatibility.md:9]:
```suggestion
- `mmcv.fileio` module, removed in PR [#2179](https://github.com/open-mmlab/mmcv/pull/2179). Fileio module from mmengine will be used wherever required.
```

File: docs/en/compatibility.md [hunk 10]
```diff
@@ -1,3 +1,53 @@
...
+
+At the same time, MMCV released [2.x](https://github.com/open-mmlab/mmcv/tree/2.x) release candidate version and will release 2.x officially version on January 1, 2023. In version 2.x, it has the following changes:
+
+(1) It removes the following components:
+
+- Removed `mmcv.fileio` module in PR [#2179](https://github.com/open-mmlab/mmcv/pull/2179) and used fileio module from mmengine wherever fileio is required.
+- Removed four modules in PR [#2216](https://github.com/open-mmlab/mmcv/pull/2216), `mmcv.runner`, `mmcv.parallel`, `mmcv. engine` and `mmcv.device`
+- Removed all classes in `mmcv.utils` (eg `Config` and `Registry`) and large Some functions in PR [#2217](https://github.com/open-mmlab/mmcv/pull/2217) , only a few functions related to mmcv are reserved.
+- Removed `mmcv.onnex`, `mmcv.tensorrt` modules and related functions in PR [#2225](https://github.com/open-mmlab/mmcv/pull/2225).
```
Comment (C1rN09) [inline docs/en/compatibility.md:12]:
```suggestion
- `mmcv.runner`, `mmcv.parallel`, `mmcv. engine` and `mmcv.device`, removed in PR [#2216](https://github.com/open-mmlab/mmcv/pull/2216).
- All classes in `mmcv.utils` (eg `Config` and `Registry`) and some large functions, removed in PR [#2217](https://github.com/open-mmlab/mmcv/pull/2217). Only a few functions related to mmcv are reserved.
- `mmcv.onnex`, `mmcv.tensorrt` and related functions, removed in PR [#2225](https://github.com/open-mmlab/mmcv/pull/2225).
```

File: docs/zh_cn/compatibility.md (hunk 7 above)
Comment (C1rN09) [reply docs/zh_cn/compatibility.md:16]: 我感觉这样更通顺一些：
> 删除了以下组件：
> - A，删除于 PR xxx
> - B，删除于 PR yyy

File: docs/en/compatibility.md (hunk 6 above)
Comment (zhouzaida) [reply docs/en/compatibility.md:16]: okay

**Review by C1rN09 — APPROVED** [review]

**Review by HAOCHENYE — APPROVED** [review]

**Pull request merged**